python-packages/smithy-core/smithy_core/deserializers.py

```diff
@@ -60,6 +60,19 @@ def read_null(self, schema: "Schema") -> None:
         """
         ...
 
+    def read_optional[
+        T
+    ](self, schema: "Schema", optional: Callable[["Schema"], T]) -> T | None:
```

Comment:
I really hate the way black formats this

Reply:
Ruff does a better job with these cases. If we manually rewrite this as the below does `black` still mess it up, or is it left alone?

```suggestion
    def read_optional[T](
        self, schema: "Schema", optional: Callable[["Schema"], T]
    ) -> T | None:
```